graylog2-server/src/main/java/org/graylog2/shared/bindings/providers/OkHttpClientProvider.java

```diff
@@ -59,6 +60,25 @@
 @Singleton
 public class OkHttpClientProvider implements Provider<OkHttpClient> {
     private static final Logger LOG = LoggerFactory.getLogger(OkHttpClientProvider.class);
+
+    static class UserAgentInterceptor implements Interceptor {
+        private final String userAgent;
+
+        public UserAgentInterceptor(String userAgent) {
+            this.userAgent = userAgent;
+        }
+
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+            Request originalRequest = chain.request();
+            Request newRequest = originalRequest.newBuilder()
+                    .header(HttpHeaders.USER_AGENT, userAgent)
+                    .build();
```

Comment:
Does this always override the User-Agent header, even if a request uses a custom one?

Reply:
Good point! Yes it does.

I will investigate how to support custom ones.

Reply:
I added support for custom user agent strings (on a per-request basis): https://github.com/Graylog2/graylog2-server/pull/21864/commits/43f371f70280263c34eb1eff8fedca1ccd1aa110

Not sure if we will need it, but good to allow this.